frontend/apps/reader/modules/readerfooter.lua

```diff
@@ -2026,7 +2039,9 @@ end
 
 function ReaderFooter:onHoldFooter()
     if self.mode == self.mode_list.off then return end
-    self.ui:handleEvent(Event:new("ShowSkimtoDialog"))
+    if self.settings.skim_widget_on_hold then
+        self.ui:handleEvent(Event:new("ShowSkimtoDialog"))
+    end
     return true
 end
```

Comment:
Hold on footer is the last handler for hold (#5789), so this is not really an issue - but logically, you'll return true only if you handled it - and let it return without true if you're skipping handling it, as something next (you don't know it there :) could be handling it.